src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1755,7 +1759,31 @@ public static Matrix4x4 Transform(Matrix4x4 value, Quaternion rotation)
         /// <returns>The transposed matrix.</returns>
         public static unsafe Matrix4x4 Transpose(Matrix4x4 matrix)
         {
-            if (Sse.IsSupported)
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                // This implementation is based on the DirectX Math Library XMMatrixTranspose method
+                // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathMatrix.inl
```

Comment:
nit: Have comment at the top.

Reply:
I don't believe the comment applies to the x64 implementation